packages/@aws-cdk/aws-s3/lib/auto-delete-objects-handler/auto-delete-objects-handler.ts

```diff
@@ -0,0 +1,189 @@
+      const https = require('https');
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      const { URL } = require('url');
+
+      const responseBody = JSON.stringify({
+        Status: respStatus,
+        Reason: reason || 'See the details in CloudWatch Log Stream: ' + ctx.logStreamName,
+        PhysicalResourceId: physResourceId || ctx.logStreamName,
+        StackId: evt.StackId,
+        RequestId: evt.RequestId,
+        LogicalResourceId: evt.LogicalResourceId,
+        Data: respData,
+      });
+
+      const parsedUrl = new URL(evt.ResponseURL);
+      const options = {
+        hostname: parsedUrl.hostname,
+        port: 443,
+        path: parsedUrl.pathname + parsedUrl.search,
+        method: 'PUT',
+        headers: {
+          'Content-Type': '',
+          'Content-Length': responseBody.length,
+        },
+      };
+
+      https.request(options)
+        .on('error', reject)
+        .on('response', (res: any) => {
+          res.resume();
+          if (res.statusCode >= 400) {
+            reject(new Error(`Server returned error ${res.statusCode}: ${res.statusMessage}`));
+          } else {
+            resolve();
+          }
+        })
+        .end(responseBody, 'utf8');
+    });
+  };
+
+  /**
+   * Recursively delete all items in the bucket
+   *
+   * @param {AWS.S3} s3 the S3 client
+   * @param {*} bucketName the bucket name
+   */
+  const emptyBucket = async function(s3: any, bucketName: string) {
+    const listedObjects = await s3.listObjectVersions({ Bucket: bucketName }).promise();
+    const contents = (listedObjects.Versions || []).concat(listedObjects.DeleteMarkers || []);
+    if (contents.length === 0) return;
+
+    let records = contents.map((record: any) => ({ Key: record.Key, VersionId: record.VersionId }));
+    await s3.deleteObjects({ Bucket: bucketName, Delete: { Objects: records } }).promise();
+
+    if (listedObjects?.IsTruncated === 'true' ) await emptyBucket(s3, bucketName);
+  };
+
+  try {
+    const bucketName = event.ResourceProperties?.BucketName;
+    if (!bucketName) throw new Error('BucketName is required');
+    physicalResourceId = `${bucketName}-${event.LogicalResourceId}`;
+
+    switch (event.RequestType) {
+      case 'Create':
+      case 'Update':
+        await report(event, context, 'SUCCESS', physicalResourceId, responseData);
+        break;
+      case 'Delete':
+        // eslint-disable-next-line @typescript-eslint/no-require-imports, import/no-extraneous-dependencies
+        const s3 = new (require('aws-sdk').S3)();
+        await emptyBucket(s3, bucketName);
+        await report(event, context, 'SUCCESS', physicalResourceId, responseData);
+        break;
+      default:
+        throw new Error(`Unsupported request type ${event.RequestType}`);
+    }
+
+    console.log('Done!');
+  } catch (err) {
+    console.log(`Caught error ${err}.`);
+    await report(event, context, 'FAILED', physicalResourceId, null, err.message);
```

Comment:
Here you pass `err.message` (a `string`) as the response data, but for successes you pass a `{}` (a `json`), are you sure this works?